File: lib/models/group.dart
class Group {
  final String id;
  final String name;
  final String? description;
  final String inviteCode;
  final String? createdBy;
  final DateTime createdAt;

  const Group({
    required this.id,
    required this.name,
    this.description,
    required this.inviteCode,
    this.createdBy,
    required this.createdAt,
  });

  factory Group.fromJson(Map<String, dynamic> json) => Group(
        id: json['id'] as String,
        name: json['name'] as String,
        description: json['description'] as String?,
        inviteCode: json['invite_code'] as String,
        createdBy: json['created_by'] as String?,
        createdAt: DateTime.tryParse(json['created_at'] as String? ?? '') ??
            DateTime.now(),
      );
}

File: PrayerWalkSwift/PrayerWalk/Models/Group.swift
// Group.swift
// PrayerWalk

import Foundation

struct Group: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let inviteCode: String
    let createdBy: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case inviteCode = "invite_code"
        case createdBy = "created_by"
        case createdAt = "created_at"
    }
}

struct GroupInsert: Encodable {
    let name: String
    let description: String
    let inviteCode: String
    let createdBy: String

    enum CodingKeys: String, CodingKey {
        case name
        case description
        case inviteCode = "invite_code"
        case createdBy = "created_by"
    }
}
